src/EnergyPlus/SetPointManager.cc

```diff
@@ -138,32 +137,34 @@ using namespace Curve;
 using Psychrometrics::PsyCpAirFnW;
 using Psychrometrics::PsyHFnTdbW;
 
-constexpr std::array<std::string_view, static_cast<int>(CtrlVarType::Num)> controlTypeName = {"Temperature",
-                                                                                              "MaximumTemperature",
-                                                                                              "MinimumTemperature",
-                                                                                              "HumidityRatio",
-                                                                                              "MaximumHumidityRatio",
-                                                                                              "MinimumHumidityRatio",
-                                                                                              "MassFlowRate",
-                                                                                              "MaximumMassFlowRate",
-                                                                                              "MinimumMassFlowRate"};
-
-constexpr std::array<std::string_view, static_cast<int>(CtrlVarType::Num)> controlTypeNameUC = {"TEMPERATURE",
-                                                                                                "MAXIMUMTEMPERATURE",
-                                                                                                "MINIMUMTEMPERATURE",
-                                                                                                "HUMIDITYRATIO",
-                                                                                                "MAXIMUMHUMIDITYRATIO",
-                                                                                                "MINIMUMHUMIDITYRATIO",
-                                                                                                "MASSFLOWRATE",
-                                                                                                "MAXIMUMMASSFLOWRATE",
-                                                                                                "MINIMUMMASSFLOWRATE"};
-
-constexpr std::array<std::string_view, static_cast<int>(ControlStrategy::Num)> strategyNamesUC = {
+constexpr std::array<std::string_view, (int)HVAC::CtrlVarType::Num> ctrlVarTypeNames = {
```

Comment:
Your opinion is more than welcome. I don't mark you as a reviewer because I figured you have enough other things to do (as do I, but that's a different story), but if you have the bandwidth and interest.

As for this, this is the only idiom (enum as an array index) where I use C-style casts, and it's mostly for readability. My presumption is that all enums are implemented as `int`s by the compiler. In my 30+ years of programming, I have never seen them implemented any other way, so the assumption is that this upcast is safe even without a check. Furthermore, if you are using an `enum` as a `std::array` index, you are already explicitly saying that you are using the `enum` as an it.

Note, I agree with you about the `EXPECT_EQ` thing. That's a different idiom, and using int casts there was a bridge too far. But I don't see a problem with them here in this idiom.

Reply:
I understand the rationale.

---

For completeness, The default is that C++ `enum class` uses integer, C enums are unsigned. In other words:

```
enum e1 {}; <=> enum e1: unsigned {}
enum class e2 {};  <=> enum e2: int (};
```

If you use the enum value for indexing into a std::array (and you can only do that because we do not have the related conversion compiler warnings enabled), you're actually saying `myarray[static_cast<std::size_t>(myEnumValue)]`. 

Which is probably fine in E+ since we typically go from Invalid = -1 to something small anyways, except if you were use the Invalid = -1 we have, and do `myarray[MyEnum::Invalid]`. static_cast<size_t>(-1) gives 18,446,744,073,709,551,616, which is definitely going to raise an out-of-bounds array indexing for our enums, but in a debug build only. In release, this is UB.

Reply:
There are actually two kinds of `enum`s in EnergyPlus. Those that come from the IDF file, and those that are used only internally.  The ones used internally should never be set to `Invalid` and probably don't even need an `Invalid` value. The ones that come from EnergyPlus probably also don't need an `Invalid` value either because @rraustad loves to remind me, those get trapped in the epJson parser and never get to EnergyPlus proper.

You can go ahead and enable the compiler warnings. In fact, I see that somehow a `-w` snuck into `flags.cmake` in `energypluslib.dir` which you probably want to get rid of while you are at it.